yarn-project/acir-simulator/src/avm/avm_memory_types.ts

```diff
@@ -261,6 +261,7 @@ export class TaggedMemory {
 
   // Truncates the value to fit the type.
   public static integralFromTag(v: bigint, tag: TypeTag): IntegralValue {
+    v = BigInt(v); // FIXME: not sure why this cast is needed, but this errors otherwise
```

Comment:
For some reason I keep getting this error when trying to run transpiled code:
```
    TypeError: Cannot mix BigInt and other types, use explicit conversions

      265 |         return new Uint16(v & ((1n << 16n) - 1n));
      266 |       case TypeTag.UINT32:
    > 267 |         return new Uint32(v & ((1n << 32n) - 1n)));
          |                            ^
      268 |       case TypeTag.UINT64:
      269 |         return new Uint64(v & ((1n << 64n) - 1n));
      270 |       case TypeTag.UINT128:
```

Reply:
Strange!